lib/trino-parquet/src/main/java/io/trino/parquet/ParquetDataSource.java

```diff
@@ -34,7 +37,11 @@
 
     Slice readFully(long position, int length);
 
-    <K> Map<K, ChunkReader> planRead(Map<K, DiskRange> diskRanges);
+    <K> Multimap<K, ChunkReader> planRead(Multimap<K, DiskRange> diskRanges);
```

Comment:
Instead of reading one big chunk per column per row group, many smaller chunks are read (based on the column index information). The disk ranges are combined in sorted order (to determine if they are contiguous or near) and the multimap ensures values preserve insertion order.

Reply:
Multimap doesn't guarantee order preservation. That's an artifact of the specific Multimap implementation. It would be clearer to use a `Map<K, List<...>>` to convey that.

Reply:
Changed to ListMultimap which is documented to preserve insertion order of values.